### ISSUE TITLE
SDK for Huawei Watch GT

### ISSUE THREAD

**Issue by alduder** [LiteOS/LiteOS#630]
Hi!

Is the LiteOS version in the Huawei Watch GT available to external developers? Is there a SDK for developers to build apps or customise the software for the Watch GT?

Thanks.

**Comment by azw413** [comment]
I notice there's no answer yet. You guys have some nice hardware but it's worthless without software. There's probably a few of us quite interested in developing 3rd party watch faces, apps and integrations but we can't do that without any docs, tools or SDK's. How about a simple guide in English covering: how to get hold of the firmware and get it running in QEMU and then how to package it up and deploy changes to the watch ? We can then build the apps whilst you're working out how to distribute them to end users.

**Comment by vipyoshi** [comment]
I agree, I want to develop stuff for my watch, kinda disappointed in the software of the Watch GT...
Not even a possibility for developer. Sadly I already read many comment of people who are disappointed in the software :/

**Comment by xsacha** [comment]
An SDK for such a watch is unrelated to the kernel the watch uses.
You can likely build apps for Lite OS by targeting this kernel, but the facility to target the GUI and deploy to the watch is unrelated to LiteOS.

As such, this issue should be closed.

**Comment by blchinezu** [comment]
Any idea if there will be a way of developing stuff for Huawei Watch GT 2 then?

**Comment by blopware** [comment]
Seconded this. If any, Google has a major business success because they allow final users and developers to tweak "their" devices.

If this is not possible with the GT2 (a really good piece of hardware, otherwise) it will eventually fall down.

**Comment by RafaelPuerta** [comment]
I want to build an application for me 😥😣

**Comment by markyin** [comment]
pls post your question in https://gitee.com/LiteOS/LiteOS

**Comment by wlad** [comment]
I know why they dont answer. They dont know english 🤣

**Comment by xsacha** [comment]
They don't answer because it is already answered. This is about the OS (LiteOS), which already allows apps. Not the watches, which have no SDK or method of deploying apps.

**Comment by xsacha** [comment]
SDK is coming December 12th AFAIK. Here is the docs:
https://developer.harmonyos.com/en/docs/documentation/doc-guides/wearable-developing-0000001052265837

**Comment by polroti** [comment]
@xsacha This is related to HarmonyOS and not LiteOS. The users (developers) of Huawei Watch GT2, GT2e & GT2 Pro cannot be benefited. The reason being no USB or WLAN connectivity options. The only possible way is via Bluetooth. I believe that LiteOS is in its early stages. We have to wait this one out.

**Comment by xsacha** [comment]
@manoj997 You'll find that HarmonyOS uses LiteOS (the kernel). https://github.com/Awesome-HarmonyOS/HarmonyOS

As far as I know, the new apps SDK is a streamed system from the phone OS. The apps do not run natively on the watch.
I understand the feature is only available for the Watch GT Pro, but as it uses the same hardware, same kernel and same UI, the restriction is arbitrary in nature. Perhaps we'll get an update on the older watches in future but more likely they want you to buy a new product.

**Comment by polroti** [comment]
@xsacha 
So is it possible to somehow run any custom app? If so, how (steps??)

**Comment by xsacha** [comment]
@manoj997 no, of course not :P Although it looks like the GT2 Pro also does not run any apps. It just streams them using an EMUI feature. It requires constant connection to an EMUI-running phone.

**Comment by polroti** [comment]
@xsacha 
I own a Huawei Watch GT2e. LiteOS. This watch is capable of ton of things. I can customize apps and install. Sadly... It's not possible. Any workarounds for this?

**Comment by nuplo** [comment]
Any update? I've just bought gt2 pro and need to write an app for the watch

**Comment by polroti** [comment]
@nuplo 
Not yet. :(

**Comment by NibbleOfAnOctet** [comment]
Apps are developed using huawei wear engine. Apparently you need an enterprise developer account. 
https://developer.huawei.com/consumer/en/hms/huawei-wearengine/

**Comment by polroti** [comment]
Update: My Huawei GT2e got the firmware update 11.0.14.26 (216 MB) and in the changelog, it says it is possible to install 3rd party apps. 
However, there are some limitations.

- I have a Huawei Nova 3i with Android 9.1 and EMUI 9.0 and in the Huawei health (12.0.12.310) there is no option for installing apps yet.
- Even if you get the install app option for other watches, there is only a handful of apps you can install and it means that you cannot build your own custom app, publish it to AppGallery and install it into your watch.
- With every potential firmware update, the noticable changes are UI changes such as icon colors and updates to weather. Other than that, you can improve things like **Support for Emojis, Support for app icons from the device. (It does not even recognize the Huawei Health App icon), Option to keep notifications even if they're viewed and possiblity to apply quick replies, Answer a call if a earbud is linked to the watch and transfer the call to the watch (This is mainly GT2e), Accuracy of compass and barometer readings, Ability to add custom always on watch faces.**

Dear LiteOS devs, please take a note of this and try to improve it for the betterment.

**Comment by xsacha** [comment]
I just got the same update.
You might need a newer app gallery

**Comment by polroti** [comment]
installed AppGallery HUAWEI AppGallery 11.6.1.200_beta and restarted the phone. Still the same :(
